### PR TITLE
Remove save button in user settings

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR removes the save button form user settings, so that all settings behave consistently.

Upon changes of notification settings, accessibility, and trip history, settings are saved immediately and a toast notification is shown as confirmation.

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
Taking this back to do some other improvements.

**Review by AdrianaCeric — APPROVED** [review]
I like the implementation! It's a lot more smooth without the save button. One comment and a question

File: lib/components/user/new-account-wizard.tsx [hunk 1]
```diff
@@ -41,20 +44,28 @@ const NewAccountWizard = ({
...
         <PageTitle title={verifyEmail} />
         <h1>{verifyEmail}</h1>
         <VerifyEmailPane {...formikProps} />
-      </>
+      </Form>
     )
   }
 
-  const { hasConsentedToTerms, notificationChannel = 'email' } = userData
+  const { hasConsentedToTerms, notificationChannel = ['email'] } = userData
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/user/new-account-wizard.tsx:68]: Correct. I also changed the type of the user data to `EditedUser`, for which enabled notifications channels are stored in an array.

File: lib/components/util/toasts.tsx [hunk 2]
```diff
@@ -8,20 +8,30 @@ import { UserSavedLocation } from '../user/types'
 // Note: the HTML for toasts is rendered outside of the IntlProvider context,
 // so intl.formatMessage and others have to be used instead of <FormattedMessage> tags.
 
+/**
+ * Helper for displaying formatted toasts.
+ */
+export function toastSuccess(title: string, description: string): void {
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/util/toasts.tsx:14]: Actually, an alert is shown if an error occurred!

File: lib/components/user/stacked-panes-with-save.tsx [hunk 3]
```diff
@@ -78,4 +65,4 @@ const StackedPaneDisplay = ({
     </>
   )
 }
-export default StackedPaneDisplay
+export default StackedPanesWithSave
```
Comment (miles-grant-ibigroup) [inline lib/components/user/stacked-panes-with-save.tsx:68]: Why do we do this instead of adding a prop? Is it formik nonsense?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
The is clearly well thought out, tested, and planned. However, there is one behavior that feels weird: can we disable the entire form during save? Otherwise it's confusing as to why the one item gets disabled. I think the other alternative would be to transform the checkbox into a spinner or something (this could be done with pure css) but as is I think it's a bit confusing.

Also, could we add a TODO to style the toasts a bit? They're a bit out of place with the rest of the UI

**Comment by binh-dam-ibigroup** [comment]
> I think the other alternative would be to transform the checkbox into a spinner or something (this could be done with pure css)

How about putting a `cursor: wait` style, with an aria-status "updating" while a single update is occurring?

**Comment by binh-dam-ibigroup** [comment]
> Also, could we add a TODO to style the toasts a bit? They're a bit out of place with the rest of the UI

I agree that toast styling could generally be improved, but I won't add a TODO for this PR.

File: lib/components/util/toasts.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdrianaCeric) [reply lib/components/util/toasts.tsx:14]: I was thinking that having the same styling for an error and a success would ensure visual consistency

[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/util/toasts.tsx:14]: Wait no I'm not a huge fan of this that might be confusing

**Comment by miles-grant-ibigroup** [comment]
> > I think the other alternative would be to transform the checkbox into a spinner or something (this could be done with pure css)
> 
> How about putting a `cursor: wait` style, with an aria-status "updating" while a single update is occurring?

Let's give it a shot!

File: lib/components/util/toasts.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdrianaCeric) [reply lib/components/util/toasts.tsx:14]: Why

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/util/toasts.tsx:14]: That's why in #992 I only applied toasts to confirmation messages and not alerts.

File: lib/components/user/stacked-panes-with-save.tsx (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/components/user/stacked-panes-with-save.tsx:68]: That's to separate the code noise brought by handling the navigation button states.

**Comment by binh-dam-ibigroup** [comment]
> > > I think the other alternative would be to transform the checkbox into a spinner or something (this could be done with pure css)
> > 
> > 
> > How about putting a `cursor: wait` style, with an aria-status "updating" while a single update is occurring?
> 
> Let's give it a shot!

See latest commits.

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
New effect looks cool! is there any way we can get the loading cursor to appear on the entire label instead of just the checkbox?

**Review by miles-grant-ibigroup — APPROVED** [review]
I would've done this with classes but this seems good enough for now! Excited for these changes to move forward it's much cleaner

**Pull request merged**